core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -85,27 +108,21 @@
     ClientTransport get();
   }
 
+  private enum State {INIT, STARTED, HALF_CLOSED, CANCELED}
```

Comment:
Yes, it is for Channels, not Calls.

Note that when I added the state checking I considered an enum. I didn't so that it was trivial to compare the logic to the documentation.